### PR TITLE
add fee to transaction as option

### PR DESCRIPTION
**Pull request by ledhed2222**
## High Level Overview of Change

Add `fee` as an option when sending XRP

### Context of Change

https://app.asana.com/0/1190398011050052/1199238264596125/f

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Just adds a new optional param when sending XRP.

Looks like a bunch of `eslint` changes from our tests are included. Just adding them too.

## Test Plan

Added one unit test to make sure that we can get transaction responses back. DID NOT verify that transactions will fail with a very small fee amount.

### CODE REVIEW

File: src/Xpring/xpring-client.ts [hunk 1]
```diff
@@ -63,23 +63,15 @@ export default class XpringClient {
...
     // Resolve the destination address to an XRP address.
     const destinationAddress = await this.payIdClient.xrpAddressForPayId(
-      destinationPayID,
+      sendXrpDetails.destination,
     )
 
     // Transact XRP to the resolved address.
     const transactionResult = await this.xrpClient.sendXrpWithDetails({
-      amount,
+      ...sendXrpDetails,
```
Comment (ledhed2222) [inline src/Xpring/xpring-client.ts:73]: these changes will make it unnecessary to change this function unless you have to pre-process a member of `SendXrpDetails`.

File: test/XRP/default-xrp-client.test.ts [hunk 2]
```diff
@@ -317,6 +307,34 @@ describe('Default XRP Client', function (): void {
...
+
+    // THEN the transaction hash exists and finds the transaction with the memo.
+    const expectedTransactionHash = Utils.toHex(
+      FakeXRPNetworkClientResponses.defaultSubmitTransactionResponse().getHash_asU8(),
+    )
+
+    assert.exists(transactionResult.hash)
+    assert.strictEqual(transactionResult.hash, expectedTransactionHash)
+  })
+
```
Comment (ledhed2222) [inline test/XRP/default-xrp-client.test.ts:337]: this is the only thing added in the test suite that wasn't an eslint fix

**Review by mvadari — COMMENTED** [review]
I think your eslint is on the wrong settings or something - when I open up the branch locally, it wants to undo all of the eslint-related changes you make here. 

Looks good other than that

File: test/XRP/default-xrp-client.test.ts [hunk 3]
```diff
@@ -317,6 +307,34 @@ describe('Default XRP Client', function (): void {
...
+    // WHEN the account makes a transaction with a maxFee
+    const transactionResult = await xrpClient.sendXrpWithDetails({
+      amount,
+      destination: destinationAddress,
+      sender: wallet,
+      maxFee,
+    })
+
+    // THEN the transaction hash exists and finds the transaction with the memo.
```
Comment (amiecorso) [inline test/XRP/default-xrp-client.test.ts:329]:
```suggestion
    // THEN the transaction hash exists and is as expected.
```

**Review by amiecorso — CHANGES_REQUESTED** [review]
I think this looks good so far.  I'm not really sure what's going on with the linting changes... since several of us have been contributing to these files recently without any such changes being invoked by the linter...  But I'm not sure how to fix it.

I think we might want to add an integration test here that 1) queries the ledger for the current required minimum fee and then 2) submits a transaction where maxfee is set higher than this (success) and potentially lower than this (failure) ?

File: test/ILP/fakes/fake-ilp-network-client.ts [hunk 4]
```diff
@@ -67,12 +67,8 @@ export class FakeIlpNetworkClientResponses {
...
    */
   public constructor(
-    public readonly getBalanceResponse: Response<
-      GetBalanceResponse
-    > = FakeIlpNetworkClientResponses.defaultGetBalanceResponse(),
-    public readonly sendResponse: Response<
-      SendPaymentResponse
-    > = FakeIlpNetworkClientResponses.defaultSendResponse(),
+    public readonly getBalanceResponse: Response<GetBalanceResponse> = FakeIlpNetworkClientResponses.defaultGetBalanceResponse(),
+    public readonly sendResponse: Response<SendPaymentResponse> = FakeIlpNetworkClientResponses.defaultSendResponse(),
```
Comment (mvadari) [inline test/ILP/fakes/fake-ilp-network-client.ts:71]: Unrelated change (also covered in #941 )

File: test/PayID/pay-id-integration.test.ts [hunk 5]
```diff
@@ -8,9 +8,7 @@ import { XrplNetwork } from 'xpring-common-js'
 const timeoutMs = 60 * 1000 // 1 minute
 
 describe('PayID Integration Tests', function (): void {
-  it('Resolve PayID to XRP - known PayID - mainnet', async function (): Promise<
-    void
-  > {
+  it('Resolve PayID to XRP - known PayID - mainnet', async function (): Promise<void> {
```
Comment (mvadari) [inline test/PayID/pay-id-integration.test.ts:11]: Same with all the changes in this file

File: test/XRP/fakes/fake-web-socket-network-client.ts [hunk 6]
```diff
@@ -36,18 +36,10 @@ export class FakeWebSocketNetworkClientResponses {
...
-    public readonly getAccountLinesResponse: Result<
-      AccountLinesResponse
-    > = FakeWebSocketNetworkClientResponses.defaultGetAccountLinesResponse(),
-    public readonly getGatewayBalancesResponse: Result<
-      GatewayBalancesResponse
-    > = FakeWebSocketNetworkClientResponses.defaultGetGatewayBalancesResponse(),
+    public readonly subscribeResponse: Result<StatusResponse> = FakeWebSocketNetworkClientResponses.defaultSubscribeResponse(),
+    public readonly unsubscribeResponse: Result<StatusResponse> = FakeWebSocketNetworkClientResponses.defaultUnsubscribeResponse(),
+    public readonly getAccountLinesResponse: Result<AccountLinesResponse> = FakeWebSocketNetworkClientResponses.defaultGetAccountLinesResponse(),
+    public readonly getGatewayBalancesResponse: Result<GatewayBalancesResponse> = FakeWebSocketNetworkClientResponses.defaultGetGatewayBalancesResponse(),
```
Comment (mvadari) [inline test/XRP/fakes/fake-web-socket-network-client.ts:42]: same

File: test/XRP/fakes/fake-xrp-network-client.ts [hunk 7]
```diff
@@ -75,21 +75,11 @@ export class FakeXRPNetworkClientResponses {
...
-      GetTransactionResponse
-    > = FakeXRPNetworkClientResponses.defaultGetTransactionResponse(),
-    public readonly getTransactionHistoryResponse: Result<
-      GetAccountTransactionHistoryResponse
-    > = FakeXRPNetworkClientResponses.defaultGetTransactionHistoryResponse(),
+    public readonly getAccountInfoResponse: Result<GetAccountInfoResponse> = FakeXRPNetworkClientResponses.defaultAccountInfoResponse(),
+    public readonly getFeeResponse: Result<GetFeeResponse> = FakeXRPNetworkClientResponses.defaultFeeResponse(),
+    public readonly submitransactionResponse: Result<SubmitTransactionResponse> = FakeXRPNetworkClientResponses.defaultSubmitTransactionResponse(),
+    public readonly getTransactionStatusResponse: Result<GetTransactionResponse> = FakeXRPNetworkClientResponses.defaultGetTransactionResponse(),
+    public readonly getTransactionHistoryResponse: Result<GetAccountTransactionHistoryResponse> = FakeXRPNetworkClientResponses.defaultGetTransactionHistoryResponse(),
```
Comment (mvadari) [inline test/XRP/fakes/fake-xrp-network-client.ts:82]: same

**Review by mvadari — CHANGES_REQUESTED** [review]
